Editing and writing a README file (#43)
redis/NRedisStack: README.md

### Redis OSS commands
You can use Redis OSS commands in the same way as you use them in [StackExchange.Redis](https://github.com/StackExchange/StackExchange.Redis).

> ### Stack Commands

Reviewer: ```suggestion
### Stack commands
```

Each module has a command class with its own commands.

> The supported modules are: [Search](https://redis.io/commands/?group=search), [Json](https://redis.io/commands/?group=json), [Graph](https://redis.io/commands/?group=graph), [TimeSeries](https://redis.io/commands/?group=timeseries), [Bloom Filter](https://redis.io/commands/?group=bf), [Cuckoo Filter](https://redis.io/commands/?group=cf), [T-Digest](https://redis.io/commands/?group=tdigest), [Count-min Sketch](https://redis.io/commands/?group=cms) and [Top-K](https://redis.io/commands/?group=topk).

Reviewer: ```suggestion
The supported modules are: [Search](https://redis.io/commands/?group=search), [Json](https://redis.io/commands/?group=json), [Graph](https://redis.io/commands/?group=graph), [TimeSeries](https://redis.io/commands/?group=timeseries), [Bloom Filter](https://redis.io/commands/?group=bf), [Cuckoo Filter](https://redis.io/commands/?group=cf), [T-Digest](https://redis.io/commands/?group=tdigest), [Count-min Sketch](https://redis.io/commands/?group=cms) and [Top-K](https://redis.io/commands/?group=topk).
```

# Usage

## 💻 Installation

Using the dotnet cli, run:

```text
dotnet add package NRedisStack
```

## 🏁 Getting started

### Starting Redis

> Before writing any code you'll need a Redis instance with the appropriate Redis modules! The quickest way to get this is with Docker:

Reviewer: ```suggestion
Before writing any code, you'll need a Redis instance with the appropriate Redis modules. The quickest way to get this is with Docker:
```

```sh
docker run -p 6379:6379 --name redis-stack redis/redis-stack:latest
```

> This launches the [redis-stack](https://redis.io/docs/stack/) an extension of Redis that adds all manner of modern data structures to Redis. You'll also notice that if you open up http://localhost:8001 you'll have access to the redis-insight GUI, a GUI you can use to visualize and work with your data in Redis.

Reviewer: ```suggestion
This launches the [Redis Stack](https://redis.io/docs/stack/), an extension of Redis that adds modern data structures to Redis. You'll also notice that if you open up http://localhost:8001 you'll have access to RedisInsight, a UI you can use to visualize and work with your data in Redis.
```

ITimeSeriesCommands ts = db.TS();
```
Then, that variable will allow you to call all the commands of that module.

> ## Basic Examples

Reviewer: ```suggestion
## Examples
```

> ### Simple Json Set Example

Reviewer: ```suggestion
### Set JSON object to Redis
```

var key = "myKey";
json.Set(key, "$", new Person() { Age = 35, Name = "Alice" });
```

> ### Index & Search Example

Reviewer: ```suggestion
### Index and search
```

Reviewer: Please provide an overview, explaining what the user is going to learn.

setup:
```csharp
using NRedisStack;
...
IDatabase db = redisFixture.Redis.GetDatabase();
ISearchCommands ft = db.FT();
IJsonCommands json = db.JSON();
```
Create an index with fields and weights:
```csharp
// FT.CREATE myIdx ON HASH PREFIX 1 doc: SCHEMA title TEXT WEIGHT 5.0 body TEXT url TEXT
ft.Create("myIndex", new FTCreateParams().On(IndexDataType.Hash)
                                         .Prefix("doc:"),
                     new Schema().AddTextField("title", 5.0)
                                 .AddTextField("body")
                                 .AddTextField("url"));
```

After you create the index, any new hash documents with the doc: prefix are automatically indexed upon creation.

> Use the HSET command to create a new hash document and add it to the index:

Reviewer: ```suggestion
To create a new hash document and add it to the index, use the HSET command:
```

```csharp
// HSET doc:1 title "hello world" body "lorem ipsum" url "http://redis.io"
db.HashSet("doc:1", new HashEntry[] { new("title", "hello world"),
                                      new("body", "lorem ipsum"),
                                      new("url", "http://redis.io") });
```
Search the index for documents that contain "hello world":
```csharp
// FT.SEARCH myIndex "hello world" LIMIT 0 10
ft.Search("myIndex", new Query("hello world").Limit(0, 10));
```

> Drop the index

Reviewer: ```suggestion
Drop the index:
```
